corehq/apps/locations/fixtures.py

```diff
@@ -56,6 +56,36 @@ def should_sync_locations(last_sync, location_db, user):
     return False
 
 
+def _gather_leaf_locations(user_location, domain):
+    """Returns all the leaf locations for which we want ancestors
+    """
+    # From the root most location we want (expand_from), we traverse down the
+    # tree until we get to the leaf-most location we want. We populate the
+    # fixture with all ancestors of these desired leafs later
+
+    location_type = user_location.location_type
+    expand_from = location_type.expand_from or location_type
+    expand_to = location_type.expand_to
+
+    if location_type.expand_from_root:
+        root = SQLLocation.root_locations(domain=domain)
+    else:
+        root = (user_location
+                .get_ancestors(include_self=True)
+                .filter(location_type=expand_from)
+                .filter(is_archived=False))
```

Comment:
just to check, this should always return exactly 0 or 1 locations right? could maybe use `.get` and `DoesNotExist` to make that explicit (not sure that's better so feel free to ignore)